great_expectations/validator/validator.py

```diff
@@ -146,37 +139,202 @@ def __init__(
             batches: The batches for which to validate.
             include_rendered_content: Whether or not to include rendered_content in the ExpectationValidationResult.
         """
+        self._data_context: Optional[AbstractDataContext] = data_context
 
-        self._data_context = data_context
-
+        self._metrics_calculator: MetricsCalculator = MetricsCalculator(
+            execution_engine=execution_engine,
+            show_progress_bars=self._determine_progress_bars(),
+        )
         execution_engine.batch_manager.reset_batch_cache()
-        self._execution_engine = execution_engine
+        self._execution_engine: ExecutionEngine = execution_engine
 
         self.load_batch_list(batch_list=batches)
 
-        self._expose_dataframe_methods = False
-
-        self._show_progress_bars = self._determine_progress_bars()
+        self._expose_dataframe_methods: bool = False
 
-        self.interactive_evaluation = interactive_evaluation
+        self.interactive_evaluation: bool = interactive_evaluation
         self._initialize_expectations(
             expectation_suite=expectation_suite,
             expectation_suite_name=expectation_suite_name,
         )
-        self._default_expectation_args = copy.deepcopy(
+        self._default_expectation_args: dict = copy.deepcopy(
```

Comment:
This looks to be `Dict[str, Union[bool, str]]`

Reply:
@NathanFarmer I am afraid it could go wider.  How about we leave it as this for now (against my immediate wish -- and yours, too).  Thanks.

Reply:
@alexsherstinsky we have `mypy` now to catch these things in case the class variable changes. Right now we know that the exact type is `Dict[str, Union[bool, str]]`.